src/main/java/com/netgrif/application/engine/importer/service/Importer.java

```diff
@@ -1054,22 +1054,33 @@ protected void createRole(Role importRole) {
             throw new IllegalArgumentException("Role ID '" + ProcessRole.ANONYMOUS_ROLE + "' is a reserved identifier, roles with this ID cannot be defined!");
         }
 
+        ProcessRole role = null;
+        if (importRole.isGlobal() == null || !importRole.isGlobal() || (importRole.isGlobal() && processRoleService.findAllByImportId(ProcessRole.GLOBAL + importRole.getId()).isEmpty())) {
```

Comment:
Maybe this if condition could be extracted to a function for better readability.